lib/store/src/csf/testing-utils/index.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  isExportStory,
+  AnyFramework,
+  AnnotatedStoryFn,
+  StoryAnnotations,
+  ComponentAnnotations,
+  Args,
+  StoryContext,
+} from '@storybook/csf';
+
+import { prepareStory } from '../prepareStory';
+import { normalizeStory } from '../normalizeStory';
+import { normalizeProjectAnnotations } from '../normalizeProjectAnnotations';
+import { HooksContext } from '../../hooks';
+import { NormalizedProjectAnnotations } from '../..';
+
+if (process.env.NODE_ENV === 'test') {
+  // eslint-disable-next-line global-require
+  const { default: addons, mockChannel } = require('@storybook/addons');
+  addons.setChannel(mockChannel());
+}
+
+export type StoryFile = {
+  default: Record<any, any>;
+  __esModule?: boolean;
+  __namedExportsOrder?: string[];
+};
+
+type PartialPlayFn = (
+  context: Partial<StoryContext> & Pick<StoryContext, 'canvasElement'>
+) => Promise<void> | void;
+
+export function composeStory<
+  TFramework extends AnyFramework = AnyFramework,
+  TArgs extends Args = Args
+>(
+  story: AnnotatedStoryFn<TFramework, TArgs> | StoryAnnotations<TFramework, TArgs>,
+  meta: ComponentAnnotations<TFramework, TArgs>,
+  globalConfig: NormalizedProjectAnnotations<TFramework> = {}
+) {
+  if (story === undefined) {
+    throw new Error('Expected a story but received undefined.');
+  }
+
+  const normalizedMeta = normalizeProjectAnnotations(meta);
+
+  const normalizedStory = normalizeStory(story.name, story, normalizedMeta);
+
+  const preparedStory = prepareStory<TFramework>(normalizedStory, normalizedMeta, globalConfig);
+
+  const defaultGlobals = Object.entries(globalConfig.globalTypes || {}).reduce(
+    (acc, [arg, { defaultValue }]) => {
+      if (defaultValue) {
+        acc[arg] = defaultValue;
+      }
+      return acc;
+    },
+    {} as Record<string, { defaultValue: any }>
+  );
```

Comment:
Can we share this code with the `GlobalsStore` somehow?

Reply:
Gotcha!